### PR TITLE
share link with clean url for community announcements

### LINKED ISSUE #908: Clean URL in announcement share link button
The current button for sharing announcements gives a link like this:

https://near.org/s/p?a=webassemblymusic.community.devhub.near&b=112244156

Which will point you directly to the post at near.org without the context of DevHUB or the community, as we can see from the screenshot

image

The preferred solution would be a cleaner URL, with the community in the path, and that the community page is shown and the post is scrolled into the view:

image

The URL should then be something like: https://devhub.near.page/community/webassemblymusic/announcements?accountId=webassemblymusic.community.devhub.near&blockHeight=112244156

### PR DESCRIPTION
**Pull request by petersalomonsen**
Share button for community announcements now creates a clean URL to the web4 gateway. The video shows clicking the copy URL button and opening the URL where you can see the format of it in the red area in the top left.

https://github.com/user-attachments/assets/ef081e5a-cec1-4c33-b763-37cde0338aff

Another difference compared to earlier is that the linked announcement now leads to the community announcements feed, and scrolls the linked announcement into view and highlights it ( earlier it just linked to the post, without any indication of which community it is about, since it was just linking to the standard post widget ). Note that all the loading indicators have to finish before scrolling the announcement into the view:

https://github.com/user-attachments/assets/97c1376a-7ea6-4156-ad2c-b7a6f80ce6aa

There is also the new file [web4browserclient.js](scripts/web4browserclient.js) which contains the script used by the web4 gateway to scroll linked posts into the view. The web4 gateway can reference this directly from any CDN that can serve github content.

fixes #908

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths Please include this PR in the race.

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: scripts/dev-gateway.mjs
Comment (Megha-Dev-19) [inline scripts/dev-gateway.mjs:21]: this is great integration.

**Review by Megha-Dev-19 — APPROVED** [review]
@petersalomonsen do you think we can highlight the post (like we do in proposal page), with black border or something, along with auto-scroll?

**Comment by Megha-Dev-19** [comment]
@race-of-sloths score 5

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@Megha-Dev-19, thank you for scoring this pull request in the Race of Sloths!

**Comment by petersalomonsen** [comment]
> @petersalomonsen do you think we can highlight the post (like we do in proposal page), with black border or something, along with auto-scroll?

Yes this should be in place now. Check the latest commit and updated video.

**Pull request merged**
